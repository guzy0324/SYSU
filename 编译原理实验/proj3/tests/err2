/* Test case 2: undelcared variable */
program err2;
class myTest
{
	declarations
		int x, y;
	enddeclarations
	method void main()
	{
	z:=x+y;
	}
}
