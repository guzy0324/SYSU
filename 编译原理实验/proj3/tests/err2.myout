Semantic Error--line: 13, symbol z: undeclared.
********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     myTest          0                            class                                                       
  2          x          1                         variable  type-tree                                            
  3          y          1                         variable  type-tree                                            
  4       main          1  proc-tree             procedure                                                      0
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,1,"myTest"]
  | +-[ClassDefOp]
  | | |             +-[DUMMYnode]
  | | |           +-[VarOp]
  | | |           | +-[STNode,3,"y"]
  | | |         +-[AddOp]
  | | |         | | +-[DUMMYnode]
  | | |         | +-[VarOp]
  | | |         |   +-[STNode,2,"x"]
  | | |       +-[AssignOp]
  | | |       | |   +-[DUMMYnode]
  | | |       | | +-[VarOp]
  | | |       | | | +-[IDNode,21,"z"]
  | | |       | +-[AssignOp]
  | | |       |   +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[STNode,4,"main"]
  | | +-[BodyOp]
  | |   |       +-[DUMMYnode]
  | |   |     +-[CommaOp]
  | |   |     | | +-[DUMMYnode]
  | |   |     | +-[TypeIdOp]
  | |   |     |   +-[INTEGERTNode]
  | |   |   +-[CommaOp]
  | |   |   | +-[STNode,3,"y"]
  | |   | +-[DeclOp]
  | |   | | |     +-[DUMMYnode]
  | |   | | |   +-[CommaOp]
  | |   | | |   | | +-[DUMMYnode]
  | |   | | |   | +-[TypeIdOp]
  | |   | | |   |   +-[INTEGERTNode]
  | |   | | | +-[CommaOp]
  | |   | | | | +-[STNode,2,"x"]
  | |   | | +-[DeclOp]
  | |   | |   +-[DUMMYnode]
  | |   +-[BodyOp]
  | |     +-[DUMMYnode]
  +-[ClassOp]
    +-[DUMMYnode]
