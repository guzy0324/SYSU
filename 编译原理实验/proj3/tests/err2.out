Semantic Error--line: 12, symbol z: undeclared.
********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     myTest          0                            class                                                       
  2          y          1                         variable  type-tree                                            
  3          x          1                         variable  type-tree                                            
  4       main          1  proc-tree             procedure                                                      0
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,1,"myTest"]
  | +-[ClassDefOp]
  | | |             +-[DUMMYnode]
  | | |           +-[VarOp]
  | | |           | +-[STNode,2,"y"]
  | | |         +-[AddOp]
  | | |         | | +-[DUMMYnode]
  | | |         | +-[VarOp]
  | | |         |   +-[STNode,3,"x"]
  | | |       +-[AssignOp]
  | | |       | |   +-[DUMMYnode]
  | | |       | | +-[VarOp]
  | | |       | | | +-[IDNode,21,"z"]
  | | |       | +-[AssignOp]
  | | |       |   +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[STNode,4,"main"]
  | | +-[BodyOp]
  | |   |       +-[DUMMYnode]
  | |   |     +-[CommaOp]
  | |   |     | | +-[DUMMYnode]
  | |   |     | +-[TypeIdOp]
  | |   |     |   +-[INTEGERTNode]
  | |   |   +-[CommaOp]
  | |   |   | +-[STNode,2,"y"]
  | |   | +-[DeclOp]
  | |   | | |     +-[DUMMYnode]
  | |   | | |   +-[CommaOp]
  | |   | | |   | | +-[DUMMYnode]
  | |   | | |   | +-[TypeIdOp]
  | |   | | |   |   +-[INTEGERTNode]
  | |   | | | +-[CommaOp]
  | |   | | | | +-[STNode,3,"x"]
  | |   | | +-[DeclOp]
  | |   | |   +-[DUMMYnode]
  | |   +-[BodyOp]
  | |     +-[DUMMYnode]
  +-[ClassOp]
    +-[DUMMYnode]
